### PR TITLE
[FEAT] Add MLPMultivariate model

### PR DESCRIPTION
**Pull request by elephaint**
This PR adds a multivariate version of the simple MLP into neuralforecast and demonstrates its application in the Multivariate_with_TSMixer example notebook. The purpose of this model is mainly for demonstration purposes, and for its simplicity.

- Added MLPMultivariate and AutoMLPMultivariate models to NeuralForecast
- Included tests in core class
- Included performance evaluation in example notebook.
- Added unit tests for loss functions to all multivariate models
- Added safeguard in BaseMultivariate for unsupported loss functions (basically all distribution loss functions)

### CODE REVIEW

File: nbs/examples/Multivariate_with_TSMixer.ipynb [hunk 1]
```diff
@@ -16,11 +16,11 @@
    "source": [
```
Comment (cchallu) [inline nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <p>I only see the forecast for the MLPMultivariate in the plot, but the code seems to be generating the forecast for all models. Can you double check the plot shows the forecasts for all models?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/938/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (cchallu) [inline nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <p>same here, I only see the MAE and MSE for MLPMultivariate.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/938/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (cchallu) [inline nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <p>remove bold for MSE in TSMixer (paper) column</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/938/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

File: nbs/models.tsmixerx.ipynb [hunk 2]
```diff
@@ -55,8 +55,6 @@
    "outputs": [],
```
Comment (cchallu) [inline nbs/models.tsmixerx.ipynb:55]: <pre class="ql-syntax" spellcheck="false">Line #236.                # x = x.reshape(batch_size, self.h, -1)               #   [B, h, n_outputs, N] -&gt; [B, h, n_outputs * N]
</pre><p>is this commented line something to review later?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/938/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (elephaint) [reply nbs/models.tsmixerx.ipynb:55]: Good catch, that's redundant, I'll remove that

File: nbs/examples/Multivariate_with_TSMixer.ipynb (hunk 1 above)
Comment (elephaint) [reply nbs/examples/Multivariate_with_TSMixer.ipynb:16]: Removed

Comment (elephaint) [reply nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <p>Should be good now</p>

Comment (elephaint) [reply nbs/examples/Multivariate_with_TSMixer.ipynb:16]: <p>Should be fixed now</p>

**Review by cchallu — APPROVED** [review]

**Pull request merged**